### PR TITLE
Fix background audio event

### PR DESCRIPTION
**Pull request by osamaasifoak**

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -26,14 +25,14 @@ import 'package:flutter/services.dart';
...
+import 'package:just_audio/just_audio.dart';
 import 'package:shared_preferences/shared_preferences.dart';
 import 'package:Medito/providers/providers.dart';
-
 import 'services/notifications/notifications_service.dart';
 
 late SharedPreferences sharedPreferences;
 late AudioPlayerNotifier audioHandler;
-
+const audioBgEvent = 'com.medito.audio.bg.event';
```
Comment (michaelcspeed) [inline lib/main.dart]: Why is this in main.dart?

File: lib/main.dart [hunk 2]
```diff
@@ -108,35 +116,54 @@ class _ParentWidgetState extends ConsumerState<ParentWidget>
...
 
     return MaterialApp.router(
       routerConfig: goRouter,
       theme: appTheme(context),
       title: ParentWidget._title,
     );
   }
+  
+  void _handleAudioCompletion(
+    WidgetRef ref,
```
Comment (michaelcspeed) [inline lib/main.dart]: I don't think this should be in main either. Just pass the extras. That will make it easier to test

File: lib/main.dart [hunk 3]
```diff
@@ -108,35 +116,54 @@ class _ParentWidgetState extends ConsumerState<ParentWidget>
...
+      audioProvider.pause();
+
+      WidgetsBinding.instance.addPostFrameCallback((_) {
+        ref.read(audioPlayPauseStateProvider.notifier).state =
+            PLAY_PAUSE_AUDIO.PAUSE;
+      });
+    }
+  }
+
+  void _handleAudioCompletionEvent(
```
Comment (michaelcspeed) [inline lib/main.dart]: This too

File: lib/main.dart (hunk 1 above)
Comment (osamaasifoak) [reply lib/main.dart]: Sorry, I have tried many different solutions to fix this issue. I wrote this line for a background service but we don't need it anymore I'm removing it now thanks @michaelcspeed

**Pull request merged**